src/middlewared/middlewared/plugins/device_/devd_events.py

```diff
@@ -51,26 +50,24 @@ def parse_devd_message(msg):
 async def devd_listen(middleware):
     global DEVD_CONNECTED
 
-    s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
-    s.connect(DEVD_SOCKETFILE)
-    # reader, writer = await asyncio.open_unix_connection(sock=s)
-    reader = s.makefile('rb')
+    reader, writer = await asyncio.open_unix_connection(path=DEVD_SOCKETFILE)
     middleware.logger.info('devd connection established')
     DEVD_CONNECTED = True
 
     while True:
-        # line = await reader.readline()
-        line = await middleware.run_in_thread(reader.readline)
+        line = await reader.readline()
         line = line.decode(errors='ignore')
         if line == "":
+            writer.close()
```

Comment:
Should we make a try/finally block that will close writer even if an exception occurs?

Reply:
Enclosed the relevant code in try/finally block.